client/src/stores/historyStore.ts

```diff
@@ -97,6 +97,42 @@ export const useHistoryStore = defineStore("historyStore", () => {
         };
     });
 
+    const pinnedHistoriesSummarizedStatus = computed(() => {
```

Comment:
This looks like something particular to the HistoryPanel, maybe you can move this computed there?

Reply:
Moved this there and exported `storedHistories` from the store to use in the history panel as well.